src/modules/dashboard/components/EditDomainDialog/EditDomainDialog.tsx

```diff
@@ -41,9 +41,32 @@ const EditDomainDialog = ({
   close,
   colony,
   colony: { colonyAddress, colonyName, domains },
-  selectedDomainId,
   isVotingExtensionEnabled,
+  ethDomainId: preselectedDomainId,
 }: Props) => {
+  const selectedDomainId = useMemo(
+    () =>
+      preselectedDomainId === 0 ||
+      preselectedDomainId === undefined ||
+      preselectedDomainId === ROOT_DOMAIN_ID
+        ? domains.find(({ ethDomainId }) => ethDomainId !== ROOT_DOMAIN_ID)
+            ?.ethDomainId
+        : preselectedDomainId,
+    [preselectedDomainId, domains],
+  );
+
+  const selectedDomain = useMemo(
+    () =>
+      domains.find(({ ethDomainId }) =>
+        preselectedDomainId === 0 ||
+        preselectedDomainId === undefined ||
+        preselectedDomainId === ROOT_DOMAIN_ID
+          ? ethDomainId !== ROOT_DOMAIN_ID
+          : ethDomainId === preselectedDomainId,
+      ),
+    [preselectedDomainId, domains],
+  );
+
```

Comment:
Okey so now youre doing almost same job twice. useMemo and everything which is inside, its not cost free. So if we already have "selectedDomain" we dont need to do almost same things for "selectedDomainId" Please refactor it to get rid of "selectedDomainId" function

Reply:
i just forgot to remove the old function. It should be removed now.